src/sentry/search/events/datasets/base.py

```diff
@@ -9,6 +9,7 @@
 
 class DatasetConfig(abc.ABC):
     custom_threshold_columns = {}
+    non_nullable_keys = set()
```

Comment:
I personally prefer `not_null`... here, quick search in code base shows we don't use `non_null`, but we have not_null in migrations:
```
src/sentry/migrations/0303_backfill_msteams_default_installation_type.py
41:        ("sentry", "0302_mep_backfill_and_not_null_snuba_query_type"),

src/sentry/migrations/0264_use_booleanfield_docintegration.py
28:        ("sentry", "0263_remove_not_null_integrationfeature"),

src/sentry/migrations/0297_backfill_alert_owners.py
59:        ("sentry", "0296_alertrule_type_not_null"),

tests/sentry/migrations/test_0297_backfill_alert_owners.py
8:#     migrate_from = "0296_alertrule_type_not_null"

tests/sentry/migrations/test_0301_mep_backfill_and_not_null_snuba_query_type.py
14:#     migrate_to = "0301_mep_backfill_and_not_null_snuba_query_type"

tests/sentry/migrations/test_0303_backfill_msteams_default_installation_type.py
6:    migrate_from = "0302_mep_backfill_and_not_null_snuba_query_type"
```

Reply:
I disagree, `non-nullable` definitely sounds more natural to me like non-nullable fields

Reply:
as i as said "i personally prefer", so rush to change anything, NOT NULL comes to my mind because of how we do in python, sql and other languages ... e.g. `The IS NOT NULL condition is used in SQL to test for a non-NULL value.`.